classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,385 @@
+		echo '<span class="comment-count-approved">' . esc_html( count( $row['comments'] ) ) . '</span>';
+		echo '</a>';
+		echo '</div>';
+	}
+
+	/**
+	 * Prints the scheduled date in a human friendly format.
+	 *
+	 * @param array $row The array representation of the current row of the table
+	 */
+	public function column_scheduled( $row ) {
+		$next_timestamp = $row['scheduled']->next()->format( 'U' );
+		echo get_date_from_gmt( gmdate( 'Y-m-d H:i:s', $next_timestamp ), 'Y-m-d H:i:s' );
+		if ( gmdate( 'U' ) > $next_timestamp ) {
+			printf( __( ' (%s ago)', 'action-scheduler' ), human_time_diff( gmdate( 'U' ), $next_timestamp ) );
+		} else {
+			echo ' (' . human_time_diff( gmdate( 'U' ), $next_timestamp ) . ')';
+		}
+	}
+
+	/**
+	 * Returns if the current action is finished or pending.
+	 *
+	 * @param ActionScheduler_Action $item
+	 */
+	protected function get_status( $item ) {
+		if ( $item->is_finished() ) {
+			return __( 'Completed', 'action-scheduler' );
+		}
+
+		return __( 'Pending', 'action-scheduler' );
+	}
+
+	/**
+	 * Bulk delete
+	 *
+	 * Deletes actions based on their ID. This is the handler for the bulk delete. It assumes the data
+	 * properly validated by the callee and it will delete the actions without any extra validation.
+	 *
+	 * @param array $ids
+	 *
+	 * @return void
+	 */
+	protected function bulk_delete( array $ids, $ids_sql ) {
+		foreach ( $ids as $id ) {
+			$this->stores->action->delete_action( $id );
+		}
+	}
+
+	/**
+	 * Implements the logic behind running an action. PP_Table_List validates the request and their
+	 * parameters are valid.
+	 */
+	protected function row_action_run( $action_id ) {
+		try {
+			ActionScheduler::runner()->process_action( $action_id );
+			$success = 1;
+		} catch ( Exception $e ) {
+			$success = 0;
+		}
+
+		set_transient( 'actionscheduler_admin_executed', compact( 'action_id', 'success' ), 30 );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function prepare_items() {
+		$this->process_bulk_action();
+
+		$this->process_row_actions();
+
+		if ( ! empty( $_REQUEST['_wp_http_referer'] ) ) {
+			// _wp_http_referer is used only on bulk actions, we remove it to keep the $_GET shorter
+			wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
+			exit;
+		}
+
+		$this->prepare_column_headers();
+
+		$per_page = $this->get_items_query_limit();
+		$query = array(
+			'per_page' => $per_page,
+			'offset'   => $this->get_items_query_offset(),
+			'status'   => $this->get_request_status(),
+			'orderby'  => 'modified',
+			'order'    => 'ASC',
+		);
+
+		$this->items = array();
+
+		$total_items = $this->stores->action->query_actions_count( $query );
+
+		foreach ( $this->stores->action->query_actions( $query ) as $id ) {
+			$item = $this->stores->action->fetch_action( $id );
+			$this->items[ $id ] = array(
+				'ID'     => $id,
+				'hook'   => $item->get_hook(),
+				'status' => $this->get_status( $item ),
```

Comment:
You might want to use `ActionScheduler::store()->get_status( $action_id )` here.